python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py

```diff
@@ -282,19 +338,71 @@ def start(self):
         self.mqtt_client.on_message = self.on_message
         self.mqtt_client.on_disconnect = self.on_disconnect
         self.mqtt_client.connect(self.broker_host, self.broker_port)
-        _LOGGER.info("Attempting to connect to MQTT broker at {}:{}...".format(self.broker_host, self.broker_port))
+        _LOGGER.info("Attempting to connect to MQTT broker at {}:{}...".format(self.broker_host,
+                                                                               self.broker_port))
 
         self.mqtt_client.loop_start()
 
     def stop(self):
         self.mqtt_client.disconnect()
         self.mqtt_client.loop_stop()
 
-    def save(self, metric, value):
+    def save(self, readings, ts):
+        if self.asset_naming == 'Topic Fragments':
+            asset = self.construct_asset_naming_topic_fragments()
+        elif self.asset_naming == 'Topic':
+            asset = self.topic
+        else:
+            asset = self.asset_name
+        if self.attach_topic_datapoint == "true":
+            readings.update({"SparkPlugB:Topic": self.topic})
         data = {
-            'asset': self.asset_name,
-            'timestamp': datetime.fromtimestamp(metric.timestamp, tz=timezone.utc
-                                                ).strftime('%Y-%m-%d %H:%M:%S.%s'),
-            'readings': {metric.name: value}
+            'asset': asset,
+            'timestamp': ts,
+            'readings': readings
         }
         async_ingest.ingest_callback(c_callback, c_ingest_ref, data)
+
+    def validate_topic(self) -> bool:
```

Comment:
Cannot a user subscribe `spBv1.0/FactoryX/#` ?

Reply:
Added comments for the validation of a topic; as of now no validation